<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>try catch 구문 사용</title>
    <script>
        function test(data){
            try{
                console.log(data+'입니다.')
                throw new Error('에러 던집니다!')
            }
            catch(e){ //e로 매개변수 사용
                console.log('catch에서 받은 에러입니다.:'+e)
                return; //finally내용을 제외하고 무조건 종료
            }
            finally{ //무조건 실행
                console.log('함수 실행을 마칩니다.')
            }
        }
        test('안녕')
    </script>
</head>
<body>
    
</body>
</html>